### ISSUE TITLE
[BITV]: Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies

### ISSUE THREAD

**Issue by JuliaKirschenheuter** [nextcloud/server#42601]
![Screenshot from 2024-01-05 17-44-58](https://github.com/nextcloud/server/assets/6078378/2fc2df0e-3854-4a1f-a3a9-2c4d9ba1a12d)

![Screenshot from 2024-01-05 17-43-14](https://github.com/nextcloud/server/assets/6078378/8c0a6639-92b1-4920-b3ef-37c719b93f6a)

**Comment by ShGKme** [comment]
More info:

Interactive elements in `menu`, including `menuitemcheckbox`, `menuitemradio` count as interactive elements itself. So a menu item representing a checkbox with a role `menuitemcheckbox` cannot have a native `checkbox` inside. All interactivity should be on the `[role=menuitemcheckbox]` element itself.

We could hide native `input` with `role=presentation` but that could be too risky.

This is sad, but we need to implement fully custom elements to have `menuitemcheckbox` or `menuitemradio` elements :(

**Comment by JuliaKirschenheuter** [comment]
Moving issue to review 6. We can think about implementation after certification
